lxd/daemon_images.go

```diff
@@ -96,7 +97,7 @@ func imageLoadStreamCache(d *Daemon) error {
 }
 
 // ImageDownload resolves the image fingerprint and if not in the database, downloads it
-func (d *Daemon) ImageDownload(op *operation, server string, protocol string, certificate string, secret string, alias string, forContainer bool, autoUpdate bool, storagePool string, preferCached bool) (*api.Image, error) {
+func (d *Daemon) ImageDownload(op *operation, server string, protocol string, certificate string, secret string, alias string, forContainer bool, autoUpdate bool, storagePool string, preferCached bool, project string) (*api.Image, error) {
```

Comment:
Maybe, or instead of a mechanical replacement of plain arguments with a struct, come up with some factoring/abstraction that makes the API more natural (just thinking loud, I'd have to look closer at the situation to propose something concrete).